crates/bevy_scene/src/scene_spawner.rs

```diff
@@ -52,234 +105,210 @@ pub enum SceneSpawnError {
 }
 
 impl SceneSpawner {
-    pub fn spawn_dynamic(&mut self, scene_handle: Handle<DynamicScene>) {
-        self.dynamic_scenes_to_spawn.push(scene_handle);
+    /// Spawn a scene.
+    ///
+    /// This will only update the world when [`scene_spawner_system`] runs, see
+    /// [`SceneSpawner::spawn_sync`] for a method with immediate world update.
+    ///
+    /// The returned [`InstanceId`] can be used later to refer to the specific
+    /// instance of the scene you spawned.
+    pub fn spawn(&mut self, scene_handle: Handle<Scene>) -> InstanceId {
+        let instance_id = InstanceId::new();
+        self.scenes_to_spawn
+            .push((scene_handle.into(), instance_id));
+        instance_id
     }
 
-    pub fn spawn(&mut self, scene_handle: Handle<Scene>) -> InstanceId {
+    /// Spawn a dynamic scene. See [`SceneSpawner::spawn`].
+    pub fn spawn_dynamic(&mut self, scene_handle: Handle<DynamicScene>) -> InstanceId {
         let instance_id = InstanceId::new();
-        self.scenes_to_spawn.push((scene_handle, instance_id));
+        self.scenes_to_spawn
+            .push((scene_handle.into(), instance_id));
         instance_id
     }
 
+    /// Spawn a scene as a child of an existing entity.
+    ///
+    /// The returned [`InstanceId`] can be used later to refer to the specific
+    /// instance of the scene you spawned.
     pub fn spawn_as_child(&mut self, scene_handle: Handle<Scene>, parent: Entity) -> InstanceId {
-        let instance_id = InstanceId::new();
-        self.scenes_to_spawn.push((scene_handle, instance_id));
+        let instance_id = self.spawn(scene_handle);
+        self.scenes_with_parent.push((instance_id, parent));
+        instance_id
+    }
+
+    /// Spawn a dynamic scene as a child of an existing entity. See
+    /// [`SceneSpawner::spawn_as_child`].
+    pub fn spawn_dynamic_as_child(
+        &mut self,
+        scene_handle: Handle<DynamicScene>,
+        parent: Entity,
+    ) -> InstanceId {
+        let instance_id = self.spawn_dynamic(scene_handle);
         self.scenes_with_parent.push((instance_id, parent));
         instance_id
     }
 
-    pub fn despawn(&mut self, scene_handle: Handle<DynamicScene>) {
-        self.scenes_to_despawn.push(scene_handle);
+    /// Despawn the provided scene. This will remove the scene and
+    /// all its related entities from the world.
+    ///
+    /// This will only update the world when [`scene_spawner_system`] runs, see
+    /// [`SceneSpawner::despawn_sync`] for a method with immediate world
+    /// update.
+    pub fn despawn(&mut self, scene_handle: Handle<Scene>) {
```

Comment:
This is one of the only API breaking change. Before, this method accepted a `Handle<DynamicScene>`, which is surprising, given that `spawn`, `spawn_sync` and `spawn_as_child` all accept a `Handle<Scene>` I think it's justified to change the API to be more consistent. The new equivalent method is `despawn_dynamic`